pupil_src/shared_modules/video_capture/base_backend.py

```diff
@@ -175,30 +175,83 @@ class Base_Manager(Plugin):
     def __init__(self, g_pool):
         super().__init__(g_pool)
 
-    def add_menu(self):
-        super().add_menu()
         from . import manager_classes
-        from pyglui import ui
 
-        self.menu_icon.order = 0.1
+        self.manager_classes = {m.__name__: m for m in manager_classes}
+
+    def on_notify(self, notification):
+        """
+        Reacts to notification:
+            ``backend.auto_select_manager``: Changes the current Manager to one that's emitted
+            ``backend.auto_activate_source``: Activates the current source via self.auto_activate_source()
+
+        Emmits notifications (indirectly):
+            ``start_plugin``: For world thread
+            ``backend.auto_activate_source``
+        """
 
-        def replace_backend_manager(manager_class):
-            if self.g_pool.process.startswith("eye"):
+        if notification["subject"].startswith("backend.auto_select_manager"):
+            target_manager_class = self.manager_classes[notification["name"]]
+            self.replace_backend_manager(target_manager_class, auto_activate=True)
+        if (
+            notification["subject"].startswith("backend.auto_activate_source")
+            and notification["name"] == self.g_pool.process
+        ):
+            self.auto_activate_source()
+
+    def replace_backend_manager(self, manager_class, auto_activate=False):
+        if self.g_pool.process.startswith("eye"):
+            if not isinstance(self.g_pool.capture_manager, manager_class):
                 self.g_pool.capture_manager.deinit_ui()
                 self.g_pool.capture_manager.cleanup()
                 self.g_pool.capture_manager = manager_class(self.g_pool)
                 self.g_pool.capture_manager.init_ui()
-            else:
+        else:
+            if not isinstance(self, manager_class):
                 self.notify_all(
                     {"subject": "start_plugin", "name": manager_class.__name__}
                 )
+        if auto_activate:
+            self.notify_all(
+                {"subject": "backend.auto_activate_source", "name": self.g_pool.process}
+            )
+
+    def auto_activate_source(self):
```

Comment:
There are backends (e.g. `realsense` and `realsense2`) that are not supposed to implement this feature.

Do you think it would be better to have a 
```python
def auto_activate_source(self):
    pass
````
in those classes?

Reply:
Yeah, Pablo already pointed this out. Then I think its ok to keep it the way it is. Maybe you can add some info in the docstring like "Implement this if your data source ....."